feat: multiplatform module for Android and iOS

adityastic opened the pull request:
Fixes #575

adityastic commented:
@iamareebjamal I plan on creating another branch to push the latest SharedFramework which we would download when compiling badge-magic-ios , let me know your suggestions on this

iamareebjamal commented:
Sounds good but why not do it in one repo

adityastic commented:
> Sounds good but why not do it in one repo

yes, that is an amazing idea, we can shift iOS directly here and maintain both the apps right here itself

adityastic commented:
@iamareebjamal we would then have to manage travis to use macOS and manually download android sdk, gradle, jdk and the rest of things inside a android docker. Let me know your thoughts, If it is worth the effort

iamareebjamal commented:
You can use environments to differentiate between both. Not sure how easy it is in Travis but here is an example of GitHub Actions - https://github.com/korlibs/klock

adityastic commented:
@iamareebjamal the catch with environments is that they run concurrently, we might want to synchronously run our project on an android image to build our iOS-framework then run it on OS X image to build iOS with the built framework.

adityastic commented:
@iamareebjamal nvm, I managed to sync it HAHA

adityastic commented:
@iamareebjamal Changes:

1. Shifted some of the logic to shared module.
2. Removed Moshi and used kotlin serializer so now we can handle same classes in any platform.
3. changed travis to build android then iOS.

Next Steps: Add iOS project tomorrow

adityastic commented:
@iamareebjamal please review

iamareebjamal commented on app/src/main/java/org/fossasia/badgemagic/adapter/SaveAdapter.kt:
```diff
@@ -10,10 +10,10 @@ import androidx.appcompat.widget.AppCompatImageView
 import com.google.android.material.chip.Chip
+import data.BadgeConfig
+import data.ConfigInfo
+import helpers.JSONHelper
```
Why not prepending org.fossasia.badgemagic and using same package structure in common module as well?

adityastic replied:
Yep, it would be perfect
